arraycontext/container/__init__.py

```diff
@@ -168,6 +173,13 @@ def is_array_container(ary: Any) -> bool:
     :returns: *True* if the instance *ary* has a registered implementation of
         :func:`serialize_container`.
     """
+
+    from warnings import warn
+    warn("is_array_container is deprecated and will be removed in 2022. "
+            "If you must know precisely whether something is an array container, "
+            "try serializing it and catch TypeError. For a cheaper option, see "
```

Comment:
If I recall correctly, the `ndarray` serialization didn't raise a `TypeError` when it found a non-object array one (maybe `ValueError`?). Should probably change that to be consistent.

Possibly out of scope, but how about introducing some `ContainerSerializationError` / `NotArrayContainerError` / `somethingError` to catch these?

Reply:
Good point! 557f9dd

Not opposed to a separate error type, but IMO out of scope for now.